File: solutions/haskell/pangram/1/src/Pangram.hs
module Pangram (isPangram) where

import Prelude (String, Bool (..), (==), (>=), (<=), (&&), (-), otherwise)
import Data.Char (ord)
import Data.Vector.Generic
import qualified Data.Vector as V

isPangram  :: String -> Bool
isPangram text  = all (== True) (listBools text) 

listBools  :: String -> V.Vector Bool
listBools []                = replicate 26 False
listBools (x:xs)
    | x >= 'a' && x <= 'z'  = (listBools xs) // [(ord x - ord 'a', True)]
    | x >= 'A' && x <= 'Z'  = (listBools xs) // [(ord x - ord 'A', True)]
    | otherwise             = listBools xs

File: solutions/kotlin/pangram/1/src/main/kotlin/Pangram.kt
object Pangram {
    fun isPangram(input: String): Boolean {
        return input.fold(Array(26){ false }) { array, ch ->
            when (ch) {
                in 'A'..'Z' -> array[ch - 'A'] = true
                in 'a'..'z' -> array[ch - 'a'] = true
            }
            array
        }.all{ it }
    }
}